tests/phpunit/tests/dependencies/styles.php

```diff
@@ -517,4 +517,98 @@ public function test_block_styles_for_viewing_with_split_styles() {
 			$GLOBALS['wp_styles']->registered['wp-block-library']->src
 		);
 	}
+
+	/**
+	 * @ticket 58394
+	 *
+	 * @covers ::wp_maybe_inline_styles
+	 */
+	public function test_wp_maybe_inline_styles() {
+		wp_register_style( 'test-handle', '/' . WPINC . '/css/classic-themes.css' );
+		wp_style_add_data( 'test-handle', 'path', ABSPATH . WPINC . '/css/classic-themes.css' );
+
+		wp_enqueue_style( 'test-handle' );
+
+		wp_maybe_inline_styles();
+
+		$this->assertFalse( $GLOBALS['wp_styles']->registered['test-handle']->src, 'Source of style should be reset to false' );
+		$this->assertNotEmpty( $GLOBALS['wp_styles']->registered['test-handle']->extra['after'], 'Source of style should set to after property' );
+	}
+
+	/**
+	 * @ticket 58394
+	 *
```

Comment:
A description explaining why `wp_filesize` should only be called once here would be useful since this is testing for a specific side effect and not unit testing the function itself. Otherwise, a refactor or enhancement made to this function could end up breaking this test and without context it would be difficult to understand why.